packages/lib/src/composables/tooltip.ts

```diff
@@ -3,41 +3,57 @@ import { computed, reactive, Ref, watch } from 'vue';
 import { getXByIndex, Scale } from './scales';
 
 import { NO_DATA, Orientation, ORIENTATIONS } from '@/utils/constants';
-import { InternalData } from '@/types/dataset';
+import { DatasetValueObject, InternalData } from '@/types/dataset';
 import { ChartOptions } from './options';
 
 export interface AnchorAttributes {
   cx: number;
   cy: number;
 }
 
-type GetHighestValuesFunction = (data: InternalData) => Array<number>;
+type GetHighestValuesFunction = (
+  data: InternalData,
+  numberOfLabels: number
+) => Array<number>;
 
 export interface TooltipConfig {
   opened: boolean;
   targetElement: Element | null;
 }
 
-function getHighestValues(data: InternalData) {
-  return data.reduce((acc, curr) => {
-    return curr.values.map((value, index) => {
+function getFilledArray(
+  originalArray: Array<number | DatasetValueObject>,
+  numberOfLabels: number
+) {
+  const difference = numberOfLabels - originalArray.length;
+  return difference > 0
+    ? [...originalArray, ...Array(difference).fill({ value: null })]
+    : originalArray;
+}
+
+function getHighestValues(data: InternalData, numberOfLabels: number) {
+  const highestValues = data.reduce((acc, curr) => {
+    return getFilledArray(curr.values, numberOfLabels).map((value, index) => {
       if (!acc[index]) return value.value ?? 0;
       return value.value > acc[index] ? value.value : acc[index];
     });
   }, [] as Array<number>);
+  return highestValues;
```

Comment:
You can return `highestValues` directly:

```suggestion
  return data.reduce((acc, curr) => {
    return getFilledArray(curr.values, numberOfLabels).map((value, index) => {
      if (!acc[index]) return value.value ?? 0;
      return value.value > acc[index] ? value.value : acc[index];
    });
  }, [] as Array<number>);
```